### PR TITLE
fix iframe buttons (go-to-dash & OCP)

### PR DESCRIPTION
**Pull request by nightpool**
integrates #1359 because of versioning/deployment/updating/merge conflict memes.

resolves #1359, #1284

image

 - adds "shaded" styles to the OCP buttons
 - adds a method for resizing the iframe
 - updates the HTML for the OCP iframe buttons to match the new button html, including tabindex and a11y

### CODE REVIEW

File: Extensions/go_to_dash.js [hunk 1]
```diff
@@ -25,28 +25,29 @@ XKit.extensions.go_to_dash = new Object({
...
+		var go_back_html = '<a href="/dashboard/2/' + next_post_id + '" class="tx-icon-button" target="_top" ' +
 			'id="xkit_gotodash" title="View on dashboard"><span class="button-label">View</span></a>';
 
 		// Remove the text from the dashboard button because otherwise the iframe
 		// overflows
 		XKit.iframe.hide_button(XKit.iframe.dashboard_button());
 
 		var place = XKit.iframe.unfollow_button().length ?
-			XKit.iframe.unfollow_button() : XKit.iframe.delete_button();
+			XKit.iframe.dashboard_button() : XKit.iframe.delete_button();
```
Comment (nightpool) [inline Extensions/go_to_dash.js]: this is a behavioral change—now go-to-dash will show up on blogs that you're not following

File: Extensions/one_click_postage.js [hunk 2]
```diff
@@ -176,44 +176,27 @@ XKit.extensions.one_click_postage = new Object({
...
+
+		function button(type, label) {
+			return `<a
+					  id="x1cpostage_in_blog_${type}"
+					  class="tx-button--with-icon tx-icon-button xkit-in-frame-button"
+					  role="button"
+					  aria-label="${label}"
+					  title="${label}"
+					  tabindex="0"
+					>
```
Comment (nightpool) [inline Extensions/one_click_postage.js]: trust me it looks better with a tabwidth of 4 :P

File: Extensions/xkit_patches.js [hunk 3]
```diff
@@ -538,41 +538,79 @@ XKit.extensions.xkit_patches = new Object({
 				button.children(".button-label").hide();
 			},
 
+            /**
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: spaces/tabs (why didn't eslint catch this?)

**Comment by AprilSylph** [comment]
The iframe isn't long enough upon page load to display everything if you have both the OCP buttons and Go-To-Dash enabled, but adding
```javascript
$(window).load(setTimeout(function(){
    $("[data-js-snowman]").click().click();
}), 100);
```
at the end of OCP's `in_blog` function fixes that, at least on my computer. I know just using a delay is messy but I can't see a proper way of making it wait for the iframe to load properl

**Comment by nightpool** [comment]
April, this is the exact thing I fixed with the size to frame function, I
just forgot to include it in go to dash as well. It works when you load go
to dash first, but not when you load it after one click postage. I'll fix
it when I'm at my computer

**Comment by AprilSylph** [comment]
That explains everything, sorry for missing that, oh my god. The iframe seemed to be giving itself a random size each time, so I got the wrong idea. Adding that to Go-To-Dash fixes it completely.
It still doesn't seem to resize at all on your own posts though, thanks staff

File: Extensions/go_to_dash.js [hunk 4]
```diff
@@ -26,27 +26,28 @@ XKit.extensions.go_to_dash = new Object({
...
 
 		// Remove the text from the dashboard button because otherwise the iframe
 		// overflows
 		XKit.iframe.hide_button(XKit.iframe.dashboard_button());
 
-		var place = XKit.iframe.unfollow_button().length ?
-			XKit.iframe.unfollow_button() : XKit.iframe.delete_button();
+		var is_following = XKit.iframe.unfollow_button().length;
+		var place = is_following ? XKit.iframe.dashboard_button() : XKit.iframe.delete_button();
 		place.before(go_back_html);
```
Comment (nightpool) [inline Extensions/go_to_dash.js]: needs "size_to_frame" afterwards.

File: Extensions/one_click_postage.js (hunk 2 above)
Comment (nightpool) [inline Extensions/one_click_postage.js]: can we just fix tabs/spaces pls

**Comment by nightpool** [comment]
😊

**Pull request merged**